src/irmin-pack/unix/gc_worker.ml

```diff
@@ -210,35 +246,16 @@ module Make (Args : Gc_args.S) = struct
        reachable from the GC commit. *)
     let live_entries =
       stats := Gc_stats.Worker.finish_current_step !stats "mapping: start";
-      (* Step 3.1 The compact list of reachable [offset, length] *)
       let live_entries = ref Live.empty in
       let register_entry ~off ~len =
         stats := Gc_stats.Worker.incr_objects_traversed !stats;
         live_entries := Live.add ~off ~len !live_entries
       in
-      (* Step 3.2 If the commit parents are referenced by offset, then include
-         the commit parents in the reachable list. The parent(s) of [commit_key]
-         must be included in the iteration because, when decoding the
-         [Commit_value.t] at [commit_key], the parents will have to be read in
-         order to produce a key for them. If the parent is referenced by hash,
-         there is no need to read it from disk, as their key is of type Indexed
-         hash. *)
-      stats :=
-        Gc_stats.Worker.finish_current_step !stats
-          "mapping: commits to reachable";
-
-      (* Step 3.3 Put the commit in the reachable file. *)
-      let off = Node_store.get_offset node_store commit_key in
-      let len = Node_store.get_length node_store commit_key in
-      register_entry ~off ~len;
-
-      (* Step 3.4 Put the nodes and contents in the reachable list. *)
       stats :=
         Gc_stats.Worker.finish_current_step !stats
           "mapping: objects to reachable";
-      iter_reachable commit node_store ~f:register_entry;
-
-      (* Step 3.5 Return *)
+      iter_reachable ~parents:false ~min_offset:Int63.zero commit_key
+        commit_store node_store ~f:register_entry;
```

Comment:
I was initially confused by `~min_offset:Int63.zero` but then realized that in this PR `iter_reachable` has two modes of calling:

1. Look at only the tree of the `commit_key` + just its parents. This corresponds to `~parents:false ~min_offset:Int63.zero`
2. Look at the objects from `commit_key` back to some offset. This corresponds to `~parents:true ~min_offset:something_not_zero`

I think this is correct based on looking at `iter_reachable`'s comment and the code usage. If so, I'm wondering if it might be more clear to have the mode explicitly modeled (this would also eliminate invalid ways of calling the function). Something like (there is probably better naming):

```ocaml
type reachability_mode =
  | Commit of commit_key
  | Traversal of { start : commit_key; stop : int63 }
```

WDYT?

Reply:
I like your new function types. :stuck_out_tongue: Kidding aside, I think that this is an improvement to the clarity of what you're modeling and that both the aliases and comments will help future readers of the code (the refactor is nice too!). :pray: